File: Java/binarysearch/ChatGPT/ChatGPT.java
public class ChatGPT {

    // Binary search method
    public static int binarySearch(int[] arr, int target) {
        int left = 0;
        int right = arr.length - 1;

        while (left <= right) {
            int mid = left + (right - left) / 2; // Avoid potential overflow

            // Check if the target is present at mid
            if (arr[mid] == target) {
                return mid;
            }

            // If target is smaller, ignore the right half
            if (arr[mid] > target) {
                right = mid - 1;
            }
            // If target is larger, ignore the left half
            else {
                left = mid + 1;
            }
        }

        // Target not found
        return -1;
    }

    public static void main(String[] args) {
        // Create and fill the array with values
        int[] arr = new int[1000];
        for (int i = 0; i < 1000; i++) {
            arr[i] = i; // Array will contain values from 0 to 999
        }

        // Test the binarySearch with a target value
        int target = 500; // Example target value
        int result = binarySearch(arr, target);

        // Optionally, output result if needed, but it's removed based on request
        // If found, result will be the index; if not, it will be -1
    }
}

File: C/binarysearch/chatGPT/chatGPT.c
#include <stdio.h>

int binarySearch(int arr[], int left, int right, int target) {
    while (left <= right) {
        int mid = left + (right - left) / 2;

        // Check if the target is present at mid
        if (arr[mid] == target) {
            return mid;
        }

        // If target is greater, ignore the left half
        if (arr[mid] < target) {
            left = mid + 1;
        }
        // If target is smaller, ignore the right half
        else {
            right = mid - 1;
        }
    }
    // Target is not present in the array
    return -1;
}

int main() {
    int arr[1000];
    for (int i = 0; i < 1000; i++) {
        arr[i] = i * 2;  // Filling array with sorted values: 0, 2, 4, ..., 1998
    }

    int target = 998;  // Example target value to search

    int result = binarySearch(arr, 0, 999, target);

    // result will hold the index if found, or -1 if not found
    if (result != -1) {
        // Target found at index `result`
    } else {
        // Target not found
    }

    return 0;
}
